src/main/java/ch/uzh/ifi/fs22/sel/group15/cozycave/server/service/ApplicationService.java

```diff
@@ -0,0 +1,182 @@
+            throw new ResponseStatusException(HttpStatus.UNAUTHORIZED,
+                    "Applications only allowed for Students");
+        }
+
+        newApplication = applicationRepository.saveAndFlush(newApplication);
+        log.debug("Created new Application: {}", newApplication);
+
+        return newApplication;
+    }
+
+
+    public @NotNull Application updateApplication(Application applicationInput) {
+        log.debug("updating Application {}", applicationInput);
+        checkIfDataIsValid(applicationInput);
+
+        Application updatedApplication = applicationRepository.findById(applicationInput.getId())
+                .orElseThrow(() -> new ResponseStatusException(HttpStatus.NOT_FOUND, "listing not found"));
+        updatedApplication.setApplicant(applicationInput.getApplicant());
+        updatedApplication.setListing(applicationInput.getListing());
+        updatedApplication.setApplication_status(applicationInput.getApplication_status());
+
+        return this.applicationRepository.saveAndFlush(updatedApplication);
+    }
+
+    public void deleteApplication(Application application) {
+        //TODO: delete all applications once a user is deleted
+        log.debug("delete Application {}", application);
+        this.applicationRepository.delete(application);
+    }
+
+    public void deleteApplication(UUID id) {
+        log.debug("delete Application {}", id);
+        this.applicationRepository.deleteById(id);
+    }
+
+    public Application decideOnApplication(Application applicationInput, User decider) {
+        log.debug("decide on Application {}", applicationInput);
+
+        checkIfDataIsValid(applicationInput);
+
+        Application updatedApplication = applicationRepository.findById(applicationInput.getId())
+                .orElseThrow(() -> new ResponseStatusException(HttpStatus.NOT_FOUND, "listing not found"));
+
+        // only the publisher of the
+        User publisher = userRepository.getOne(applicationInput.getListing().getPublisher().getId());
+        if (publisher != decider
+                && (!decider.getRole().greaterEquals(Role.ADMIN))) {
+            throw new ResponseStatusException(HttpStatus.UNAUTHORIZED,
+                    "You're not the publisher of the listing to decide upon the application");
+        }
+        updatedApplication.setApplication_status(applicationInput.getApplication_status());
+
+        // decline all other applications automatically
+        if (updatedApplication.getApplication_status() == ApplicationStatus.ACCEPTED) {
+            for (Application deniedApplication : applicationRepository.findByListing_Id(updatedApplication.getListing().getId())) {
+                if (deniedApplication.getId() == updatedApplication.getId()) {
+                    continue;
+                }
+                deniedApplication.setApplication_status(ApplicationStatus.DENIED);
+                applicationRepository.saveAndFlush(deniedApplication);
```

Comment:
you could just save everything and flush outside of the loop. this would be faster on large amount of applications